### PR TITLE
improve disposer implementation

### PR DESCRIPTION
**Pull request by FMS-Cat**
- enable `vrm.dispose()` on dnd examples
- Refactor `disposer` a bit

### CODE REVIEW

File: src/vrm/utils/disposer.ts [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
+
 import * as THREE from 'three';
 
-function disposeMaterial(material: any): void {
+function disposeMaterial(material: THREE.Material): void {
   Object.keys(material).forEach((propertyName) => {
-    if (!!material[propertyName] && typeof material[propertyName].dispose === 'function') {
-      material[propertyName].dispose();
+    const value = (material as any)[propertyName];
+    if (value && typeof value === 'object' && value.isTexture) {
```
Comment (yomotsu) [inline src/vrm/utils/disposer.ts]: Can we ommit `typeof value === 'object'` ?
```suggestion
    if (value && value.isTexture) {
```

File: src/vrm/utils/disposer.ts [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
+      const texture = (material as any)[propertyName] as THREE.Texture;
```
Comment (yomotsu) [inline src/vrm/utils/disposer.ts]: maybe?
```suggestion
      const texture = value as THREE.Texture;
```

File: src/vrm/utils/disposer.ts (hunk 1 above)
Comment (FMS-Cat) [reply src/vrm/utils/disposer.ts]: might be? I have no clue that it is omittable

File: src/vrm/utils/disposer.ts (hunk 2 above)
Comment (FMS-Cat) [reply src/vrm/utils/disposer.ts]: definitely

File: src/vrm/utils/disposer.ts (hunk 1 above)
Comment (yomotsu) [reply src/vrm/utils/disposer.ts]: I think it's okay.
The line of the newer code is doing just like an optional-chaining. I think.

Comment (FMS-Cat) [reply src/vrm/utils/disposer.ts]: I just had no confidence I can index every truthy.
I know I cannot index `undefined` and `null`

Comment (FMS-Cat) [reply src/vrm/utils/disposer.ts]: yes I think I can everything other than `undefined` or `null` , my bad
I need the optional chaining so bad...... 😩

**Comment by yomotsu** [comment]
LGTM!

**Pull request merged**